Cabal/Distribution/Types/TestSuite.hs

```diff
@@ -43,11 +44,12 @@ instance Semigroup TestSuite where
         testInterface = combine  testInterface,
         testBuildInfo = combine  testBuildInfo
     }
-        where combine   field = field a `mappend` field b
-              combine' f = case (f a, f b) of
-                        ("", x) -> x
-                        (x, "") -> x
-                        (x, y) -> error "Ambiguous values for test field: '"
+        where combine  field = field a `mappend` field b
+              combine' field = case ( unUnqualComponentName $ field a
```

Comment:
I'd much prefer it if you could add a function like "isNullUnqualComponentName" to factor this type of test. It would be more efficient too since unUnqualComponentName requires serializing to String.